### ISSUE TITLE
Multi-user installer: don't fail if /nix exists

### ISSUE THREAD

**Issue by Mic92** [NixOS/nix#2407]
On some space constraint systems, where I sneak in nix I have /nix on a different partition.
The current installer however does not let me install nix, when `/nix` is mounted and present:

```
---- oh no! --------------------------------------------------------------------
There are some relics of a previous installation of Nix at /nix, and
this scripts assumes Nix is _not_ yet installed. Please delete the old
Nix installation and start again.
Uninstalling nix:
1. Delete the files Nix added to your system:
  sudo rm -rf /etc/nix /nix /root/.nix-profile /root/.nix-defexpr /root/.nix-channels /home/mbailleu/.nix-profile /home/mbailleu/.nix-defexpr /home/mbailleu/.nix-channels
and that is it.
We'd love to help if you need it.
If you can, open an issue at https://github.com/nixos/nix/issues
```

Proposed solution: Check if /nix/ is not empty instead.

**Comment by Mic92** [comment]
cc @grahamc what do you think about that?

**Comment by Mic92** [comment]
Oh, weird I think systemd mounts my /nix partition during the installation.

**Comment by abathur** [comment]
This can be closed; the core issue was fixed in #3054.

**Comment by abathur** [comment]
For the record, I noticed after commenting on this one that despite #3054 getting merged, the commit that fixed this issue hasn't made it into 2.3-maintenance.

So, this is "fixed", but you'll still see this on current installers until that is either cherry picked or there's a new release from master.
